builder/dockerfile/copy.go

```diff
@@ -405,26 +407,35 @@ func performCopyForInfo(dest copyInfo, source copyInfo, options copyFileOptions)
 		// is a symlink
 		destPath = filepath.Join(destPath, filepath.Base(source.path))
 	}
-	return copyFile(archiver, srcPath, destPath)
+	return copyFile(archiver, srcPath, destPath, options.chownPair)
 }
 
-func copyDirectory(archiver *archive.Archiver, source, dest string) error {
+func copyDirectory(archiver *archive.Archiver, source, dest string, chownPair idtools.IDPair) error {
+	destExists, err := isExistingDirectory(dest)
+	if err != nil {
+		return errors.Wrapf(err, "failed to query destination path")
+	}
 	if err := archiver.CopyWithTar(source, dest); err != nil {
 		return errors.Wrapf(err, "failed to copy directory")
 	}
-	return fixPermissions(source, dest, archiver.IDMappings.RootPair())
+	if !destExists {
+		// if the destination didn't exist, we need to special case chown it
+		// to the requested --chown flags as provided to the original command
+		if err := os.Lchown(dest, chownPair.UID, chownPair.GID); err != nil {
+			return errors.Wrapf(err, "failed to chown target directory")
+		}
```

Comment:
This (and the `isExistingDirectory()` call above) is unnnecessary. It's already handled by `fixPermissions()`